processmanager/manager.go

```diff
@@ -346,3 +349,65 @@ func (pm *ProcessManager) MaybeNotifyAPMAgent(
 
 	return serviceName
 }
+
+func hashFrameCacheKey(fk frameCacheKey) uint32 {
+	return uint32(uint64(fk.Frame.File) + uint64(fk.Frame.Lineno))
```

Comment:
`File` is a hash (so pretty random distribution). `Lineno` is the virtual address within the file. Hashing just by summing which should be enough to get things on different bucket. But better hash algorithms are also welcome.